includes/class-wc-payments-account.php

```diff
@@ -187,6 +187,76 @@ public function get_is_live() {
 		return ! empty( $account ) && isset( $account['is_live'] ) ? $account['is_live'] : null;
 	}
 
+	/**
+	 * Gets the various anti-fraud services that must be included on every WCPay-related page.
+	 *
+	 * @return array Assoc array. Each key is the slug of a fraud service that must be incorporated to every page, the value is service-specific config for it.
+	 */
+	public function get_fraud_services_config() {
+		$account = $this->get_cached_account_data();
+		if ( empty( $account ) || ! isset( $account['fraud_services'] ) ) {
+			// This was the default before adding new anti-fraud providers, preserve backwards-compatibility.
+			return [ 'stripe' => [] ];
+		}
+		$services_config = $account['fraud_services'];
+		foreach ( $services_config as $service_id => $config ) {
+			$prepare_func = [ $this, "prepare_fraud_${service_id}_config" ];
+			if ( is_callable( $prepare_func ) ) {
+				$services_config[ $service_id ] = call_user_func( $prepare_func, $config );
+			}
+		}
+		return $services_config;
+	}
+
+	/**
+	 * Adds site-specific config needed to initialize the SIFT anti-fraud JS.
+	 *
+	 * @param array $config Associative array with the SIFT-related configuration returned from the server.
+	 *
+	 * @return array|NULL Assoc array, ready for the client to consume, or NULL if the client shouldn't enqueue this script.
+	 */
+	private function prepare_fraud_sift_config( $config ) {
+		// The server returns both production and sandbox beacon keys. Use the sandbox one if test mode is enabled.
+		if ( WC_Payments::get_gateway()->is_in_test_mode() ) {
+			$config['beacon_key'] = $config['sandbox_beacon_key'];
+		}
+		unset( $config['sandbox_beacon_key'] );
+
+		$wpcom_blog_id = $this->payments_api_client->get_blog_id();
+		if ( ! $wpcom_blog_id ) {
+			// Don't enqueue the SIFT script if Jetpack hasn't been connected yet.
+			return null;
+		}
+
+		if ( is_user_logged_in() ) {
+			if ( is_admin() ) {
+				$config['user_id'] = $this->get_stripe_account_id();
+			} else {
+				$config['user_id'] = $wpcom_blog_id . '_' . get_current_user_id();
```

Comment:
Should we try to use the WCPay (Stripe) `Customer ID` if possible?

Reply:
I think so. That's what's going to be used as the `$user_id` for the order tracking, so they'll match.

Do you think the order creation/update should fallback to `$wpcom_blog_id . '_' . WC()->session->get_customer_id()` if it is a guest checkout? We can get the `customer_id` for guest checkouts as well, but it won't match the JS `$user_id`.

Reply:
If possible this will be great for future reference though possibly also something to do in a separate issue.

Reply:
If possible this will be great for future reference though possibly also something to do in a separate issue.

Reply:
For posterity's sake, after discussing it we decided:
- If a user is logged-in but doesn't have a Stripe Customer ID (i.e. has never purchased anything) it's treated as a "guest". `user_id` will be empty and `session_id` will be `{BLOG_ID}_{LOCAL_USER_ID}`.
- If a user has a Stripe Customer ID, we use that as the `user_id`.
- For a logged-out user, the `session_id` will be `{BLOG_ID}_{COOKIE_MD5_HASH}`.
- When a guest checkout happens, we'll send the `session_id` on the `create_customer` WCPay call, so the server can link the newly created customer with the session.
- When a logged-out user logs in, if the user has a Stripe Customer ID associated, we send a `link_user_to_session` call to the server (not implemented yet).